tools/AutomationTestFramework/RunAutomationTests.ps1

```diff
@@ -0,0 +1,128 @@
+# ----------------------------------------------------------------------------------
+    # Copyright Microsoft Corporation
+    # Licensed under the Apache License, Version 2.0 (the "License");
+    # you may not use this file except in compliance with the License.
+    # You may obtain a copy of the License at
+    # http://www.apache.org/licenses/LICENSE-2.0
+    # Unless required by applicable law or agreed to in writing, software
+    # distributed under the License is distributed on an "AS IS" BASIS,
+    # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    # See the License for the specific language governing permissions and
+    # limitations under the License.
+# ----------------------------------------------------------------------------------
+
+param (
+    [Parameter(Mandatory = $true)]
+    [string] $spPassword,
+    [bool] $createPackages = $true,
+    [bool] $uploadPackages = $true,
+    [bool] $processRunbooks = $true,
+    [bool] $waitForResults = $true
+)
+
+. "$PSScriptRoot\Management\PackageGenerator.ps1"
+. "$PSScriptRoot\Management\ModuleUploader.ps1"
+. "$PSScriptRoot\Management\RunbookProcessor.ps1"
+
+try {
+    $srcPath = "$PSScriptRoot\..\..\src"
+    $projectList = @('Compute', 'Network', 'Profile', 'Resources', 'Sql', 'Storage', 'Websites')
```

Comment:
Any reason this module list shouldn't be a parameter with a default value?

Reply:
Again, another thing that was only 'bubbled up' from my refactoring, but not exposed. Additionally, I'd like to have the list of signed modules also be exposed, so you could pick-and-choose what signed modules you actually want uploaded to Azure Automation.

Honestly, the default should be __everything__, so that all RPs are searched for tests (marked with SmokeTest). This type of list should be an Include filter. We can discuss further if you'd like.

Reply:
Yeah, thinking about it, we would need to derive the list of uploaded modules before it would be dsafe to add new modules, so we can skip this for now.